ambry-rest/src/main/java/com.github.ambry.rest/NettyServer.java

```diff
@@ -49,27 +50,20 @@
   private final NettyMetrics nettyMetrics;
   private final EventLoopGroup bossGroup;
   private final EventLoopGroup workerGroup;
-  private final RestRequestHandler requestHandler;
-  private final PublicAccessLogger publicAccessLogger;
-  private final RestServerState restServerState;
+  private final ArrayList<Map.Entry<String, ChannelHandler>> channelHandlers;
```

Comment:
You bring up a good point - we shouldn't have anything with the same keys which means a `Map` is ideal here.